lgc/builder/YCbCrConvertHandler.h

```diff
@@ -0,0 +1,272 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  YCbCrConvertHandler.h
+ * @brief LLPC header file: contains the definition of LLPC utility class YCbCrConvertHandler.
+ ***********************************************************************************************************************
+ */
+
+#pragma once
+
+#include "BuilderImpl.h"
+#include "YCbCrAddressHandler.h"
+#include "vkgcDefs.h"
```

Comment:
As I know, LGC doesn't reference VKGC definitions. Please check this with Tim.

Reply:
Yes, I found this issue when trying to use a struct which is defined in `vkgcDefs.h` and is being used both in xgl and lgc.

@trenouf, How should we deal with this problem?

Reply:
As Rex says, LGC code should not include any front-end files. In fact, I don't understand how you got this to work, as it should not even be on the include path.

What in vkgcDefs.h are you using?

Reply:
I'm need to use Vkgc::SamplerYCbCrConversionMetaData here.

Reply:
@trenouf Any suggestions?

Reply:
How about lgc/interface/lgc/CommonDefs.h? Could that be included from all the necessary places, including xgl?

Reply:
Ah, I see how you got it to work -- you added extra include paths into lgc/CMakeLists.txt. You can't do that; LGC cannot refer to anything outside LGC and LLVM.

When doing the LGC separation, I did try having a common file LgcDefs.h that contains things that are part of both the LLPC and LGC interfaces. But I ended up deciding that was actually less maintainable than having two copies of a small number of things with static_asserts that they are the same at point of use, or two slightly different versions of things with conversion code (particularly ShaderStage).

So I think we should just have a second copy of this struct, probably in YCbCrConvertHandler.h itself, with a comment in each of the two locations referring to the other. The non-LGC copy then doesn't particularly need to be in vkgcDefs.h does it? It could be in the part of the driver that constructs the converting sampler.

In the copy of the struct in LGC, could you please rephrase the comments that mention VK_COMPONENT_SWIZZLE to not rely on a Vulkan name? Thanks.